analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -3691,3 +3691,109 @@ void TestMethod()
         }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/8495
+public class Repro_8495
+{
+    private readonly static object Lock = new object();
+
+    public bool WithLock(bool a, bool b)
+    {
+        bool flag = true;
+        lock (Lock)
+        {
+            if (a)
+            {
+                return true;
+            }
+            if (b)
+            {
+                flag = false;
+            }
+        }
+        if (flag)           // Noncompliant FP
+        {
+            return true;    // Secondary FP
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    public bool WithUsing(bool a, bool b)
+    {
+        bool flag = true;
+        using (var ms = new MemoryStream())
+        {
+            if (a)
+            {
+                return true;
+            }
+            if (b)
+            {
+                flag = false;
+            }
+        }
+        if (flag)           // Noncompliant FP
+        {
+            return true;    // Secondary FP
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    public bool WithTryFinally(bool a, bool b)
+    {
+        bool flag = true;
+        try
+        {
+            if (a)
+            {
+                return true;
+            }
+            if (b)
+            {
+                flag = false;
+            }
+        }
+        finally
+        {
+            a.ToString();
```

Comment:
Educational: what is the purpose of this invocation?

Reply:
To make sure that the `finally` will not get optimized away.

Reply:
It's a good question!